irispy/spectrograph.py

```diff
@@ -486,8 +486,11 @@ def read_iris_spectrograph_level2_fits(filenames, spectral_windows=None):
     """
     if type(filenames) is str:
         filenames = [filenames]
+    if isinstance(filenames, list):
+        raise TypeError("filenames must be a string or list of strings giving"
+                        " the filenames to read.")
```

Comment:
Hi @tiagopereira.  Won't this raise an error when ```filenames``` is a ```list``` even though ```filenames``` is allowed to be a ```list```?  Are you trying check that all elements are strings in the case that ```filenames``` is a list?